File: leetcode080.py
class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        k = 1
        ls = len(nums)
        i = 0
        while i < ls-1:
            if nums[i] == nums[i+1]:
                k += 1
                if k == 3:
                    nums.pop(i+1)
                    ls = ls - 1
                    k = k -1
                    continue

                i += 1
            else:
                k = 1
                i = i + 1
        return len(nums)

if __name__ == "__main__":
    s = Solution()
    print(s.removeDuplicates([1,1,1,2,2,3]))

File: leetcodec++/leetcode080.cpp
//
// Created by 黄敏辉 on 2019-07-29.
//

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int count = 1;
        for (int i = 1;i < nums.size(); ){
            if (nums[i] != nums[i-1]) {
                count = 1;
                i++;
            }
            else if (nums[i] == nums[i-1] && count == 1) {
                count += 1;
                i++;
            }
            else if (nums[i] == nums[i-1] && count == 2)
                nums.erase(nums.begin() + i);
        }
        return nums.size();
    }
};

